app/src/main/java/org/oppia/android/app/resumelesson/ResumeLessonActivity.kt

```diff
@@ -0,0 +1,103 @@
+package org.oppia.android.app.resumelesson
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import org.oppia.android.app.activity.InjectableAppCompatActivity
+import org.oppia.android.app.home.RouteToExplorationListener
+import org.oppia.android.app.player.exploration.ExplorationActivity
+import javax.inject.Inject
+
+/** Activity that allows the user to resume a saved exploration. */
+class ResumeLessonActivity : InjectableAppCompatActivity(), RouteToExplorationListener {
+
+  @Inject
+  lateinit var resumeLessonActivityPresenter: ResumeLessonActivityPresenter
+  private var internalProfileId: Int = -1
+  private lateinit var topicId: String
+  private lateinit var storyId: String
+  private lateinit var explorationId: String
+  private var backflowScreen: Int = -1
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    activityComponent.inject(this)
+    internalProfileId = intent.getIntExtra(RESUME_LESSON_ACTIVITY_PROFILE_ID_ARGUMENT_KEY, -1)
+    topicId =
+      checkNotNull(intent.getStringExtra(RESUME_LESSON_ACTIVITY_TOPIC_ID_ARGUMENT_KEY)) {
+        "Expected topic ID to be included in intent for ResumeLessonActivity."
+      }
+    storyId =
+      checkNotNull(intent.getStringExtra(RESUME_LESSON_ACTIVITY_STORY_ID_ARGUMENT_KEY)) {
+        "Expected story ID to be included in intent for ResumeLessonActivity."
+      }
+    explorationId =
+      checkNotNull(intent.getStringExtra(RESUME_LESSON_ACTIVITY_EXPLORATION_ID_ARGUMENT_KEY)) {
+        "Expected exploration ID to be included in intent for ResumeLessonActivity."
+      }
+    backflowScreen = intent.getIntExtra(RESUME_LESSON_ACTIVITY_BACKFLOW_SCREEN_KEY, -1)
+    resumeLessonActivityPresenter.handleOnCreate(
+      internalProfileId,
+      topicId,
+      storyId,
+      explorationId,
+      backflowScreen,
+    )
+  }
+
+  // TODO(#1655): Re-restrict access to fields in tests post-Gradle.
+  companion object {
+    private const val RESUME_LESSON_ACTIVITY_PROFILE_ID_ARGUMENT_KEY =
+      "ResumeLessonActivity.profile_id"
```

Comment:
Instead of `profileId`/ `ProfileId`/`PROFILE_ID`, etc. use `internalProfileId`/`InternalProfileId`/`INTERNAL_PROFILE_ID` everywhere.

Basically when its `Int` we use `internalProfileId` as `ProfileId` is actually a model in our proto.

Reply:
done